src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -141,48 +150,51 @@ internal static string GetDefaultProvider(int dwType)
         /// <summary>
         /// Creates a new key container
         /// </summary>
-        private static void CreateCSP(CspParameters parameters, bool randomKeyContainer, ref SafeProvHandle safeProvHandle)
+        private static void CreateCSP(CspParameters parameters, bool randomKeyContainer, out SafeProvHandle safeProvHandle)
         {
-            //don't persist in CSP if this is random key container. 
-            safeProvHandle.PersistKeyInCsp = randomKeyContainer ? false : true;
-
             uint dwFlags = (uint)CryptAcquireContextFlags.CRYPT_NEWKEYSET;
             if (randomKeyContainer)
             {
                 dwFlags |= (uint)CryptAcquireContextFlags.CRYPT_VERIFYCONTEXT;
             }
-            int ret = OpenCSP(parameters, dwFlags, ref safeProvHandle);
+
+            SafeProvHandle hProv;
+            int ret = OpenCSP(parameters, dwFlags, out hProv);
             if (S_OK != ret)
             {
+                hProv.Dispose();
```

Comment:
Okay, well; then that means it was created in antiquity, so I'm cool with deleting the logging todos :)